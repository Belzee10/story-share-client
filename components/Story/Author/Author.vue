<template>
  <div class="d-flex align-end">
    <div>
      <v-img
        :src="avatar || require('~/assets/images/no-image.jpg')"
        :alt="name"
        :lazy-src="require('~/assets/images/image-loading.jpeg')"
        :height="size.height"
        :width="size.width"
        class="avatar"
      ></v-img>
    </div>
    <span class="text-capitalize ml-1 name accent--text text--darken-4">{{
      name
    }}</span>
  </div>
</template>

<script>
export default {
  name: 'Author',
  props: {
    avatar: {
      type: String,
      default: ''
    },
    name: {
      type: String,
      default: ''
    },
    size: {
      type: Object,
      default: () => ({
        height: 20,
        width: 20
      })
    }
  }
};
</script>

<style lang="scss" scoped>
.avatar {
  border-radius: 50%;
}
.name {
  font-size: 13px;
  text-decoration: underline;
}
</style>
