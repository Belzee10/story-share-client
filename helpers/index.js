/**
 * transform the response from express
 * remove values from exclude
 * change the key "_id" by "id"
 * @param {Object} item
 * @param {Array} exclude
 */
export const transformResponse = (item, exclude = ['__v', 'created_at']) => {
  Object.keys(item).forEach(key => {
    if (exclude.includes(key)) delete item[key];
  });
  if (item._id) {
    item.id = item._id;
    delete item._id;
  }
  return item;
};
