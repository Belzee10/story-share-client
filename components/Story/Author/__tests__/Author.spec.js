import { createWrapper } from '~/config/tests.helpers';
import Author from '../Author';

describe('Author.vue', () => {
  test('should render correctly', () => {
    const wrapper = createWrapper(Author);
    expect(wrapper.element).toMatchSnapshot();
  });
});
