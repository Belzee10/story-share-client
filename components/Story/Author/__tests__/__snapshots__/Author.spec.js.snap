// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author.vue should render correctly 1`] = `
<div
  class="d-flex align-end"
>
  <div>
    <div
      aria-label=""
      class="v-responsive v-image avatar"
      style="height: 20px; width: 20px;"
    >
      <div
        class="v-responsive__content"
      />
    </div>
  </div>
   
  <span
    class="text-capitalize ml-1 name accent--text text--darken-4"
  >
    
  </span>
</div>
`;
